phovea_server/util.py

```diff
@@ -42,9 +43,52 @@ def to_json(obj, *args, **kwargs):
   kwargs['ensure_ascii'] = False
 
   # Pandas JSON module has been deprecated and removed. UJson cannot convert numpy arrays, so it cannot be used here. The JSON used here does not support the `double_precision` keyword.
+  if isinstance(obj, dict) or isinstance(obj, list):
```

Comment:
the complex cases work perfectly now as far as I can see. however here you also want to check if it's a number to call the `_handle_nan_values` otherwise it won't be called for the simple case of `float('nan')`